File: kdb-example/src/main/java/com/t13max/kdb/table/_Table.java
package com.t13max.kdb.table;

import com.t13max.kdb.Kdb;
import com.t13max.kdb.bean.Human;
import com.t13max.kdb.cache.CoroutineSafeCache;
import com.t13max.kdb.cache.DefaultTableCache;
import com.t13max.kdb.conf.KdbConf;
import com.t13max.kdb.storage.IStorage;
import com.t13max.kdb.storage.MongoStorage;

/**
 * 表 自动生成
 *
 * @author t13max
 * @since 18:42 2025/7/15
 */
public class _Table extends Tables {

    private final static _Table INSTANCE = new _Table(Kdb.getInstance().getConf());

    Table<Human> humans;

    public _Table(KdbConf kdbConf) {
        super(kdbConf);

        IStorage storage = new MongoStorage();

        humans = new Table<>(Human.class, kdbConf.tableConfMap.get("humans"), new CoroutineSafeCache<Human>(new DefaultTableCache<>()), storage);

    }

    public static _Table getInstance() {
        return INSTANCE;
    }
}

File: kdb-example/src/main/kotlin/com/t13max/kdb/table/_Table.kt
package com.t13max.kdb.table

import com.t13max.kdb.Kdb
import com.t13max.kdb.bean.Human
import com.t13max.kdb.cache.CoroutineSafeCache
import com.t13max.kdb.cache.DefaultTableCache
import com.t13max.kdb.conf.KdbConf
import com.t13max.kdb.storage.IStorage
import com.t13max.kdb.storage.MongoStorage

/**
 *
 * @author t13max
 * @since 12:52 2025/8/11
 */
class _Table : Tables{
    private static
    val INSTANCE: _Table = _Table(Kdb.getInstance().getConf())

    var humans: Table<Human>? = null

    fun _Table(kdbConf: KdbConf) {
        super(kdbConf)

        val storage: IStorage = MongoStorage()

        humans = Table<V?>(
            Human::class.java,
            kdbConf.tableConfMap.get("humans")!!,
            CoroutineSafeCache<Human>(DefaultTableCache<Human>()),
            storage
        )
    }

    fun getInstance(): _Table {
        return INSTANCE
    }
}